app/src/format.py

```diff
@@ -27,64 +28,100 @@ def format_guru_cards(
+        if chunk_with_score.score < docs_shown_min_score:
+            logger.info(
+                "Skipping chunk with score less than %f: %s",
+                docs_shown_min_score,
+                chunk_with_score.chunk.document.name,
             )
-            if existing_doc:
-                existing_doc.max_score = max(existing_doc.max_score, chunk_with_score.score)
-            else:
-                documents_with_scores.append(DocumentWithMaxScore(document, chunk_with_score.score))
+            continue
 
-    # Sort the list by score
-    documents_with_scores.sort(key=lambda d: d.max_score, reverse=True)
+        if document in documents:
+            documents[document].append(chunk_with_score)
+        else:
+            documents[document] = [chunk_with_score]
 
-    # Only return the top docs_shown_max_num documents
-    return [d.document for d in documents_with_scores[:docs_shown_max_num]]
+    return documents
 
 
 def format_bem_documents(
     docs_shown_max_num: int,
     docs_shown_min_score: float,
-    chunks_with_scores: Sequence[ChunkWithScore],
+    chunks_with_scores: list[ChunkWithScore],
 ) -> str:
     documents = _get_bem_documents_to_show(
         docs_shown_max_num, docs_shown_min_score, chunks_with_scores
     )
-    formatted_documents = "".join([f"<li>{document.name}</li>" for document in documents])
-    return f"<h3>Source(s)</h3><ul>{formatted_documents}</ul>"
+
+    return _format_to_accordion_group_html(documents)
+
+
+def _format_to_accordion_html(document: Document, score: float) -> str:
+    global _accordion_id
+    _accordion_id += 1
+    similarity_score = f"<p>Similarity Score: {str(score)}</p>"
+
+    return f"""
+    <div class="usa-accordion" id=accordion-{_accordion_id}>
+        <h4 class="usa-accordion__heading">
+            <button
+                type="button"
+                class="usa-accordion__button"
+                aria-expanded="false"
+                aria-controls="a-{_accordion_id}"
+                >
+                <a href='https://link'>{document.name}</a>
+            </button>
+        </h4>
+        <div id="a-{_accordion_id}" class="usa-accordion__content usa-prose" hidden>
+            {"<p>Summary: " + document.content.strip() if document.content else ""}</p>
+            {similarity_score}
+        </div>
+    </div>"""
+
+
+def _format_to_accordion_group_html(documents: OrderedDict[Document, list[ChunkWithScore]]) -> str:
+    global _accordion_id
+    _accordion_id += 1
+    html = "<h3>Source(s)</h3>"
+    internal_citation = ""
+    for document in documents:
+        for index, chunk in enumerate(documents[document], start=1):
+            formatted_chunk = re.sub(r"\n+", "\n", chunk.chunk.content).strip()
+            formatted_chunk = f"<p>Summary: {formatted_chunk} </p>" if formatted_chunk else ""
+            citation = f"<h4>Citation #{index} (score: {chunk.score})</h4>"
+            similarity_score = f"<p>Similarity Score: {str(chunk.score)}</p>"
```

Comment:
```suggestion
            similarity_score = f"<p>Similarity Score: {chunk.score}</p>"
```

Reply:
Not type casting it to the float to str, causes issues with the text overflowing out of the citations
<img width="811" alt="Screenshot 2024-08-12 at 3 16 39 PM" src="https://github.com/user-attachments/assets/9e87fc72-b1b3-442d-bfcb-a8fbd00b63d0">